src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1358,12 +1392,15 @@ internal Collection<FunctionDescriptor> GetFunctionDescriptors(IEnumerable<Funct
                         }
                     }
 
-                    ValidateFunction(descriptor, httpFunctions);
-
                     if (descriptor != null)
                     {
+                        ValidateFunction(descriptor, httpFunctions);
                         functionDescriptors.Add(descriptor);
                     }
+                    else
+                    {
+                        throw new Exception("Could not find a valid provider. Check that you have the correct language provider installed and enabled.");
```

Comment:
Could not find a valid language descriptor?

Reply:
I think provider makes sense. User can't control the descriptors, they can control the providers. Eventually, we might only have 1 descriptor, which has pluggable providers.